### PR TITLE
Make FilteredQuery a forbidden API.

### PR DESCRIPTION
**Pull request by jpountz**
This commit makes FilteredQuery a forbidden API and also removes some more usage
of the Filter API. There are some remaining code using filters for parent/child
queries but I'm not touching this as they are already being refactored in #6511.

### CODE REVIEW

File: src/main/java/org/elasticsearch/index/mapper/core/ShortFieldMapper.java [hunk 1]
```diff
@@ -209,14 +208,11 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
...
-    public Filter nullValueFilter() {
+    public Query nullValueFilter() {
         if (nullValue == null) {
             return null;
         }
-        return new QueryWrapperFilter(NumericRangeQuery.newIntRange(names.indexName(), precisionStep,
-                nullValue.intValue(),
-                nullValue.intValue(),
-                true, true));
+        return new ConstantScoreQuery(termQuery(nullValue, null));
```
Comment (rmuir) [inline src/main/java/org/elasticsearch/index/mapper/core/ShortFieldMapper.java]: How come nullValueFilter (no idea what this method is doing, its name is horrible, are we sure we really need it??????) is CSQ(term()) here for short/float but NRQ for long. Why the inconsistency?

**Comment by rmuir** [comment]
Overall looks good. i added one comment for a WTF.

**Comment by jpountz** [comment]
Good catch, this was a left-over on my end, I pushed a new commit.

**Comment by rmuir** [comment]
thanks, looks good to me. i still wonder what these methods are doing. We should investigate in a future PR. If you don't know, then I think thats a good sign we should axe them.

**Comment by jpountz** [comment]
If I read https://www.elastic.co/guide/en/elasticsearch/guide/current/_dealing_with_null_values.html#_missing_filter correctly, the goal of this option is to be able to make a difference between a document that is missing a value and a document that was explicitely provided `null` as a value.

**Comment by rmuir** [comment]
Worthless.

**Pull request merged**